ethereum/datastructures/src/test/java/tech/pegasys/artemis/datastructures/operations/AttestationDataTest.java

```diff
@@ -61,12 +64,83 @@ void shouldNotBeProcessableBeforeFirstSlotOfTargetEpoch() {
             UnsignedLong.valueOf(1),
             UnsignedLong.ZERO,
             Bytes32.ZERO,
-            new Checkpoint(UnsignedLong.ONE, Bytes32.ZERO),
+            new Checkpoint(ONE, Bytes32.ZERO),
             target);
 
     assertThat(data.getEarliestSlotForProcessing()).isEqualTo(target.getEpochStartSlot());
   }
 
+  @Test
+  public void shouldNotBeAbleToIncludeInBlockBeforeSlotAfterCreationSlot() {
+    final AttestationData data =
+        new AttestationData(
+            UnsignedLong.valueOf(60),
+            UnsignedLong.ZERO,
+            Bytes32.ZERO,
+            new Checkpoint(ONE, Bytes32.ZERO),
+            new Checkpoint(ONE, Bytes32.ZERO));
+
+    assertThat(data.canIncludeInBlockAtSlot(UnsignedLong.valueOf(60))).isFalse();
+    assertThat(data.canIncludeInBlockAtSlot(UnsignedLong.valueOf(61))).isTrue();
```

Comment:
Due to the MIN_ATTESTATION_INCLUSION_DELAY comment I made above, this line would need to change as well to:
```suggestion
    assertThat(data.canIncludeInBlockAtSlot(UnsignedLong.valueOf(60).plus(MIN_ATTESTATION_INCLUSION_DELAY))).isTrue();
```